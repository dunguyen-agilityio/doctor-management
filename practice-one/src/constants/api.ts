import Constants from 'expo-constants';

export const API_ENTITIES = {
  FOODS: 'foods',
  FAVORITES: 'favorites',
};

export const API_ENDPOINT = Constants.expoConfig?.extra?.apiEndpoint;
