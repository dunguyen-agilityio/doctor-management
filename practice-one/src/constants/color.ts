export const COLOR = {
  PRIMARY: '#5c5c5c',
  WHITE: '#fff',
  PINK: '#91c788',
  LIGHT_GRAY: '#ececec',
  GRAY: '#7e7e7e',
  RED: '#ea1502',
  PURPLE: '#9059a7',
  BLACK: '#000000',
  LIGHT_BLACK: '#330600',
  WARNING: '#da6317',
  LIGHT_PURPLE: '#fff3f1',
  GREEN: '#27ae60',
  ORANGE: '#d89e4a',
  YELLOW: '#f2c94c',
  LIGHT_GREEN_10: 'rgba(28, 195, 121, 0.1)',
  DARK_GREEN: '#1cc379',
  SECONDARY: '#ff8473',
  BLACK_BEAN: '#330600',
  LIGHT_GREEN: '#91c788',
  GREY: '#e0e0e0',
  TERTIARY: '#696969',
};
