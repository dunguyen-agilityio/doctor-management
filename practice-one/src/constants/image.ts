export const BLUR_HASH =
  '|rF?hV%2WCj[ayj[a|j[az_NaeWBj@ayfRayfQfQM{M|azj[azf6fQfQfQIpWXofj[ayj[j[fQayWCoeoeaya}j[ayfQa{oLj?j[WVj[ayayj[fQoff7azayj[ayj[j[ayofayayayj[fQj[ayayj[ayfjj[j[ayjuayj[';

export enum APP_ICONS {
  EMPTY = require('@assets/images/empty.webp'),
  NOT_FOUND = require('@assets/images/not-found.webp'),
  LOGO = require('@assets/images/logo.webp'),
  ARROW_RIGHT_BOLD = require('@assets/icons/arrow-right-bold.webp'),
  ARROW_LEFT = require('@assets/icons/arrow-left.webp'),
  QUESTION = require('@assets/icons/question.webp'),
  SEARCH = require('@assets/icons/icon-search.webp'),
  FAVORITE = require('@assets/icons/favorite.webp'),
  FAVORITE_FILL = require('@assets/icons/favorite-fill.webp'),
  HOME = require('@assets/icons/home.webp'),
  HOME_FILL = require('@assets/icons/home-fill.webp'),
  SEARCH_MENU = require('@assets/icons/search.webp'),
  SEARCH_MENU_FILL = require('@assets/icons/search-fill.webp'),
}
