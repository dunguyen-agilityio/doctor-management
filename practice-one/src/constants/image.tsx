import { Image, StyleSheet } from 'react-native';

export const EmptyImage = () => (
  <Image
    source={require('@/assets/images/empty_2x.png')}
    style={styles.empty}
  />
);

export const NotFoundImage = () => (
  <Image
    source={require('@/assets/images/not_found_2x.png')}
    style={styles.notFound}
    testID="not-found-image"
  />
);

export const Logo = () => (
  <Image source={require('@/assets/images/logo.png')} style={styles.loading} />
);

export const ArrowICon = () => (
  <Image source={require('@/assets/icons/arrow.png')} />
);

export const QuestionIcon = () => (
  <Image source={require('@/assets/icons/question.png')} />
);

export const SearchIcon = () => (
  <Image source={require('@/assets/icons/iconsearch.png')} />
);

export const FavoriteMenu = ({ isFill = false }: { isFill?: boolean }) => {
  const path = isFill
    ? require('@/assets/icons/favoritefill.png')
    : require('@/assets/icons/favorite.png');
  return <Image source={path} />;
};

export const HomeMenu = ({ isFill = false }: { isFill?: boolean }) => {
  const path = isFill
    ? require('@/assets/icons/homefill.png')
    : require('@/assets/icons/home.png');
  return <Image source={path} />;
};

export const SearchMenu = ({ isFill = false }: { isFill?: boolean }) => {
  const path = isFill
    ? require('@/assets/icons/searchfill.png')
    : require('@/assets/icons/search.png');
  return <Image source={path} />;
};

export const SplashImage = () => (
  <Image
    source={require('@/assets/images/pattern.png')}
    style={styles.topImage}
  />
);

export const styles = StyleSheet.create({
  empty: {
    height: 96,
    width: 108,
  },
  notFound: {
    width: 96,
    height: 96,
  },
  loading: {
    width: 122,
    height: 122,
  },
  topImage: {
    width: '100%',
    resizeMode: 'cover',
  },
});
