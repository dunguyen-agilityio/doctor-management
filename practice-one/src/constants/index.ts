export * from './api';
export * from './color';
export * from './category';
export * from './image';
export * from './screens';
