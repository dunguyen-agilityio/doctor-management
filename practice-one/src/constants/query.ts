export const QUERY_KEYS = {
  FOOD: 'food-list',
  FOOD_FAVORITE: 'food-list-favorite',
  FOOD_BY_ID: (id: string) => `food-${id}`,
};

export const PAGE_SIZE = 10;
