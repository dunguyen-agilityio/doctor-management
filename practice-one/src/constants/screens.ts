export enum ROUTES {
  HOME = 'Home-Screen',
  SEARCH = 'Search-Screen',
  DETAIL = 'Detail-Screen',
  FAVORITE = 'Favorite-Screen',
  ROOT = 'Root-Screen',
  SPLASH = 'SPLASH-Screen',
}
