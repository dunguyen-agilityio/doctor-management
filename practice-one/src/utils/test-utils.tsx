import { RenderOptions, render } from '@testing-library/react-native';

import { NavigationContainer } from '@react-navigation/native';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';

const AllTheProviders = ({ children }: React.PropsWithChildren) => {
  const queryClient = new QueryClient({
    defaultOptions: { queries: { retry: false } },
  });

  return (
    <QueryClientProvider client={queryClient}>
      <NavigationContainer>{children}</NavigationContainer>
    </QueryClientProvider>
  );
};

const customRender = (ui: React.ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react-native';

export { customRender as render };
