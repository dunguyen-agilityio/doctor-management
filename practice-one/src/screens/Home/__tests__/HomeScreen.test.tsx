import { useNavigation } from '@react-navigation/native';

import HomeScreen from '@/screens/Home';

import { CATEGORIES, ROUTES } from '@/constants';

import { fireEvent, render } from '@/utils/test-utils';

describe('HomeScreen', () => {
  const mockNavigate = jest.fn();
  const [{ name: categoryName, id: categoryId }] = CATEGORIES;

  beforeEach(() => {
    (useNavigation as jest.Mock).mockReturnValue({ navigate: mockNavigate });
  });

  it('renders all main components', () => {
    const { getByText, getByPlaceholderText } = render(<HomeScreen />);

    expect(getByText('All Food')).toBeTruthy(); // Checks header title in FoodsContainer
    expect(getByPlaceholderText('Search for healthy food')).toBeTruthy(); // Search input
  });

  it('navigates to search screen when search input is focused', () => {
    const { getByPlaceholderText } = render(<HomeScreen />);

    fireEvent(getByPlaceholderText('Search for healthy food'), 'focus');

    expect(mockNavigate).toHaveBeenCalledWith(ROUTES.SEARCH, {
      autoFocus: true,
    });
  });

  it('navigates to search screen with category filter when a category is selected', () => {
    const { getByText } = render(<HomeScreen />);

    fireEvent.press(getByText(categoryName));

    expect(mockNavigate).toHaveBeenCalledWith(ROUTES.SEARCH, {
      category: categoryId,
    });
  });
});
