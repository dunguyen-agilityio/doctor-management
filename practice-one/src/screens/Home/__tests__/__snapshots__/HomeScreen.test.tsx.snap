// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen match snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 20,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#5c5c5c",
            "fontFamily": "Manrope",
            "fontSize": 23,
            "fontWeight": "800",
            "lineHeight": 32,
            "textTransform": undefined,
          },
          undefined,
        ]
      }
    >
      Want to eat
healthy Food?
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      role="button"
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#00000021",
          "borderRadius": "100%",
          "height": 35,
          "justifyContent": "center",
          "width": 35,
        }
      }
      testID="question-icon"
    >
      <Image
        source={
          {
            "testUri": "../../../assets/icons/question.png",
          }
        }
      />
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#fff3f1",
        "borderRadius": 15,
        "marginHorizontal": 16,
        "marginTop": 14,
        "position": "relative",
        "zIndex": 1,
      }
    }
  >
    <TextInput
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="Search for healthy food"
      placeholderTextColor="#ff8473"
      style={
        {
          "color": "#000000",
          "fontSize": 13,
          "fontWeight": "400",
          "paddingHorizontal": 60,
          "paddingVertical": 14,
        }
      }
      testID="search-input"
    />
    <View
      style={
        {
          "left": 21,
          "position": "absolute",
          "top": "50%",
          "transform": [
            {
              "translateY": -10,
            },
          ],
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../assets/icons/iconsearch.png",
          }
        }
      />
    </View>
  </View>
  <View
    style={
      {
        "height": 26,
        "marginTop": 16,
      }
    }
  >
    <RCTScrollView
      collapsable={false}
      contentContainerStyle={
        {
          "gap": 16,
        }
      }
      handlerTag={1}
      handlerType="NativeViewGestureHandler"
      horizontal={true}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      pagingEnabled={true}
      showsHorizontalScrollIndicator={false}
      waitFor={
        [
          {
            "current": null,
          },
        ]
      }
    >
      <View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#ececec",
              "borderRadius": 8,
              "justifyContent": "center",
              "marginLeft": 16,
              "marginRight": 0,
              "opacity": 1,
              "paddingHorizontal": 10,
              "paddingVertical": 4,
            }
          }
        >
          <Text
            style={
              [
                {
                  "color": "#5c5c5c",
                  "fontFamily": "Manrope",
                  "fontSize": 13,
                  "fontWeight": "400",
                  "lineHeight": 18,
                  "textTransform": undefined,
                },
                false,
              ]
            }
          >
            Breakfast
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#ececec",
              "borderRadius": 8,
              "justifyContent": "center",
              "marginLeft": 0,
              "marginRight": 0,
              "opacity": 1,
              "paddingHorizontal": 10,
              "paddingVertical": 4,
            }
          }
        >
          <Text
            style={
              [
                {
                  "color": "#5c5c5c",
                  "fontFamily": "Manrope",
                  "fontSize": 13,
                  "fontWeight": "400",
                  "lineHeight": 18,
                  "textTransform": undefined,
                },
                false,
              ]
            }
          >
            Dinner
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#ececec",
              "borderRadius": 8,
              "justifyContent": "center",
              "marginLeft": 0,
              "marginRight": 0,
              "opacity": 1,
              "paddingHorizontal": 10,
              "paddingVertical": 4,
            }
          }
        >
          <Text
            style={
              [
                {
                  "color": "#5c5c5c",
                  "fontFamily": "Manrope",
                  "fontSize": 13,
                  "fontWeight": "400",
                  "lineHeight": 18,
                  "textTransform": undefined,
                },
                false,
              ]
            }
          >
            Dry fruits
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#ececec",
              "borderRadius": 8,
              "justifyContent": "center",
              "marginLeft": 0,
              "marginRight": 0,
              "opacity": 1,
              "paddingHorizontal": 10,
              "paddingVertical": 4,
            }
          }
        >
          <Text
            style={
              [
                {
                  "color": "#5c5c5c",
                  "fontFamily": "Manrope",
                  "fontSize": 13,
                  "fontWeight": "400",
                  "lineHeight": 18,
                  "textTransform": undefined,
                },
                false,
              ]
            }
          >
            Lunch
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#ececec",
              "borderRadius": 8,
              "justifyContent": "center",
              "marginLeft": 0,
              "marginRight": 0,
              "opacity": 1,
              "paddingHorizontal": 10,
              "paddingVertical": 4,
            }
          }
        >
          <Text
            style={
              [
                {
                  "color": "#5c5c5c",
                  "fontFamily": "Manrope",
                  "fontSize": 13,
                  "fontWeight": "400",
                  "lineHeight": 18,
                  "textTransform": undefined,
                },
                false,
              ]
            }
          >
            Vegetables
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#ececec",
              "borderRadius": 8,
              "justifyContent": "center",
              "marginLeft": 0,
              "marginRight": 0,
              "opacity": 1,
              "paddingHorizontal": 10,
              "paddingVertical": 4,
            }
          }
        >
          <Text
            style={
              [
                {
                  "color": "#5c5c5c",
                  "fontFamily": "Manrope",
                  "fontSize": 13,
                  "fontWeight": "400",
                  "lineHeight": 18,
                  "textTransform": undefined,
                },
                false,
              ]
            }
          >
            Greenish
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#ececec",
              "borderRadius": 8,
              "justifyContent": "center",
              "marginLeft": 0,
              "marginRight": 0,
              "opacity": 1,
              "paddingHorizontal": 10,
              "paddingVertical": 4,
            }
          }
        >
          <Text
            style={
              [
                {
                  "color": "#5c5c5c",
                  "fontFamily": "Manrope",
                  "fontSize": 13,
                  "fontWeight": "400",
                  "lineHeight": 18,
                  "textTransform": undefined,
                },
                false,
              ]
            }
          >
            Non-Veg
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#ececec",
              "borderRadius": 8,
              "justifyContent": "center",
              "marginLeft": 0,
              "marginRight": 16,
              "opacity": 1,
              "paddingHorizontal": 10,
              "paddingVertical": 4,
            }
          }
        >
          <Text
            style={
              [
                {
                  "color": "#5c5c5c",
                  "fontFamily": "Manrope",
                  "fontSize": 13,
                  "fontWeight": "400",
                  "lineHeight": 18,
                  "textTransform": undefined,
                },
                false,
              ]
            }
          >
            Fast Food
          </Text>
        </View>
      </View>
    </RCTScrollView>
  </View>
  <View
    style={
      {
        "marginTop": 16,
      }
    }
  >
    <View
      style={
        {
          "height": 266.8,
        }
      }
    >
      <View
        style={
          [
            {
              "display": "flex",
            },
            undefined,
          ]
        }
      >
        <View
          collapsable={false}
          onLayout={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          style={
            [
              {
                "overflow": "hidden",
              },
              {
                "height": "100%",
                "width": 750,
              },
              {
                "height": 266.8,
              },
              {
                "flexDirection": "row",
              },
            ]
          }
        />
      </View>
    </View>
    <View
      style={
        [
          {
            "alignSelf": "center",
            "justifyContent": "space-between",
            "minHeight": 20,
            "minWidth": 20,
          },
          {
            "flexDirection": "row",
          },
          {
            "alignItems": "center",
            "flexDirection": "row",
            "gap": 10,
            "height": 20,
            "justifyContent": "center",
            "marginTop": 23,
            "width": 30,
          },
        ]
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "overflow": "hidden",
              "transform": [
                {
                  "rotateZ": "0deg",
                },
              ],
            },
            {
              "backgroundColor": "rgba(255, 132, 115, 0.5)",
              "borderRadius": 16,
              "height": 8,
              "width": 12,
            },
            {
              "backgroundColor": undefined,
              "borderRadius": undefined,
              "height": undefined,
              "transform": [],
              "width": undefined,
            },
          ]
        }
      />
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "overflow": "hidden",
              "transform": [
                {
                  "rotateZ": "0deg",
                },
              ],
            },
            {
              "backgroundColor": "rgba(255, 132, 115, 0.5)",
              "borderRadius": 16,
              "height": 8,
              "width": 12,
            },
            {
              "backgroundColor": undefined,
              "borderRadius": undefined,
              "height": undefined,
              "transform": [],
              "width": undefined,
            },
          ]
        }
      />
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "overflow": "hidden",
              "transform": [
                {
                  "rotateZ": "0deg",
                },
              ],
            },
            {
              "backgroundColor": "rgba(255, 132, 115, 0.5)",
              "borderRadius": 16,
              "height": 8,
              "width": 12,
            },
            {
              "backgroundColor": undefined,
              "borderRadius": undefined,
              "height": undefined,
              "transform": [],
              "width": undefined,
            },
          ]
        }
      />
    </View>
  </View>
  <View
    style={
      {
        "gap": 20,
        "marginTop": 20,
        "paddingHorizontal": 20,
      }
    }
    testID="food-skeleton"
  >
    <View
      style={
        {
          "backgroundColor": "#E0E0E0",
          "height": 16,
          "marginTop": 10,
          "width": 80,
        }
      }
    />
    <View
      style={
        {
          "flexDirection": "row",
          "gap": 20,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderColor": "#DBDBDB",
            "borderRadius": 20,
            "borderWidth": 1,
            "height": 192,
            "justifyContent": "space-around",
            "minWidth": 154,
            "paddingBottom": 18,
            "paddingHorizontal": 27,
            "paddingTop": 17,
            "paddingVertical": 17,
          }
        }
      >
        <View
          style={
            {
              "backgroundColor": "#E0E0E0",
              "borderRadius": 8,
              "height": 60,
              "width": 60,
            }
          }
        />
        <View
          style={
            {
              "backgroundColor": "#E0E0E0",
              "height": 14,
              "marginTop": 10,
              "width": "80%",
            }
          }
        />
        <View
          style={
            {
              "backgroundColor": "#E0E0E0",
              "height": 14,
              "marginTop": 10,
              "width": "80%",
            }
          }
        />
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderColor": "#DBDBDB",
            "borderRadius": 20,
            "borderWidth": 1,
            "height": 192,
            "justifyContent": "space-around",
            "minWidth": 154,
            "paddingBottom": 18,
            "paddingHorizontal": 27,
            "paddingTop": 17,
            "paddingVertical": 17,
          }
        }
      >
        <View
          style={
            {
              "backgroundColor": "#E0E0E0",
              "borderRadius": 8,
              "height": 60,
              "width": 60,
            }
          }
        />
        <View
          style={
            {
              "backgroundColor": "#E0E0E0",
              "height": 14,
              "marginTop": 10,
              "width": "80%",
            }
          }
        />
        <View
          style={
            {
              "backgroundColor": "#E0E0E0",
              "height": 14,
              "marginTop": 10,
              "width": "80%",
            }
          }
        />
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderColor": "#DBDBDB",
            "borderRadius": 20,
            "borderWidth": 1,
            "height": 192,
            "justifyContent": "space-around",
            "minWidth": 154,
            "paddingBottom": 18,
            "paddingHorizontal": 27,
            "paddingTop": 17,
            "paddingVertical": 17,
          }
        }
      >
        <View
          style={
            {
              "backgroundColor": "#E0E0E0",
              "borderRadius": 8,
              "height": 60,
              "width": 60,
            }
          }
        />
        <View
          style={
            {
              "backgroundColor": "#E0E0E0",
              "height": 14,
              "marginTop": 10,
              "width": "80%",
            }
          }
        />
        <View
          style={
            {
              "backgroundColor": "#E0E0E0",
              "height": 14,
              "marginTop": 10,
              "width": "80%",
            }
          }
        />
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderColor": "#DBDBDB",
            "borderRadius": 20,
            "borderWidth": 1,
            "height": 192,
            "justifyContent": "space-around",
            "minWidth": 154,
            "paddingBottom": 18,
            "paddingHorizontal": 27,
            "paddingTop": 17,
            "paddingVertical": 17,
          }
        }
      >
        <View
          style={
            {
              "backgroundColor": "#E0E0E0",
              "borderRadius": 8,
              "height": 60,
              "width": 60,
            }
          }
        />
        <View
          style={
            {
              "backgroundColor": "#E0E0E0",
              "height": 14,
              "marginTop": 10,
              "width": "80%",
            }
          }
        />
        <View
          style={
            {
              "backgroundColor": "#E0E0E0",
              "height": 14,
              "marginTop": 10,
              "width": "80%",
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;
